### PR TITLE
fix: opening and closing the menu by clicking the caret

### LINKED ISSUE #61: Clicking on open select caret does not close the menu
Clicking the caret opens the menu, it should also close it when the menu is open. However currently it just makes the select flicker but does not close it. 

_Originally posted by @agazso in https://github.com/diete-design/diete.design/pull/57#pullrequestreview-2016516897_

### PR DESCRIPTION
**Pull request by hlavkat**
resolve #61

### CODE REVIEW

**Review by vojtechsimetka — CHANGES_REQUESTED** [review]
Thanks, this works but I have few comments:

- It probably should be `button` instead of `div` (for accessibility reasons)
- if it is a `button` then `onkeypress` will work correctly out of the box, otherwise it should have the same code as the `onclick` handler but with `enter` key

File: src/lib/components/ui/select/select.svelte [hunk 1]
```diff
@@ -21,19 +21,25 @@
...
+	// Close the select when user clicks outside, when user clicks on the tab button
 	$effect(() => {
 		function closeMenu() {
 			if (store.open) store.open = false
 		}
 
 		window.addEventListener('click', closeMenu)
+		window.addEventListener('keydown', (e) => {
+			if (e.key === 'Tab') closeMenu()
+		})
```
Comment (vojtechsimetka) [inline src/lib/components/ui/select/select.svelte:38]: do we need this on `window` or can we have it just on the `input` element?

**Review by vojtechsimetka — APPROVED** [review]
Mostly good from my end, WDYT @agazso ?

**Review by agazso — APPROVED** [review]
So now it closes the menu, so functionally it works.

It still leaves the input active (focused) which is debatable, but at least it is consistent with what happens when you click the input twice (once to open and again to close). 

However there is a slight difference, when the caret is clicked there is an intermediate unfocused state with a small animation and the background turning grey. With clicking the input twice there is no intermediate state, and that makes clicking the caret feeling "glitchy".  This also happens when the input is focused and the caret is clicked to open it.

This seems like a minor detail, but these small inconsistencies makes the difference for people to feel the experience as smooth vs. buggy. Feel free to address this in another issue/PR.

**Pull request merged**